### ISSUE TITLE
System tests for global historic metrics for graphs

### ISSUE THREAD

**Issue by killbotXD** [0chain/system_test#420]
system tests for following endpoints that return global historic metrics for graphs
part of 30 new REST API endpoints 

#### global historic average metrics for graphs
`/graph-write-price`
`/graph-total-challenge-pools`

#### global historic total metrics for graphs
`/graph-allocated-storage`
`/graph-used-storage`
`/graph-total-locked`
`/graph-total-minted`

#### global historic difference metrics for graphs
`/graph-total-staked`
`/graph-challenges`
`/graph-token-supply`

system tests for following endpoints that return global totals
part of 30 new REST API endpoints

`/total-minted`
`/total-blobber-capacity`
`/average-write-price`
`/total-stored-data`
`/total-staked`
`/total-successful-challenges`
`/total-total-challenges`
`/total-allocated-storage`

system tests for following endpoints that return per blobber historic difference metric for graphs
part of 30 new REST API endpoints

`/graph-blobber-service-charge`
`/graph-blobber-challenges-passed`
`/graph-blobber-challenges-completed`
`/graph-blobber-inactive-rounds`

system tests for following endpoints that return per blobber historic averaged metrics for graphs
part of 30 new REST API endpoints

`/graph-blobber-write-price`
`/graph-blobber-capacity`
`/graph-blobber-allocated`
`/graph-blobber-saved-data`
`/graph-blobber-read-data`
`/graph-blobber-offers-total`
`/graph-blobber-unstake-total`
`/graph-blobber-total-stake`
`/graph-blobber-challenges-open`

Done in https://github.com/0chain/0chain/pull/1475.

**Comment by devyetii** [comment]
The endpoint `/total-stored-data` is no more found. Also I can't find it needed in any of the UIs (I mean the total stored data throughout all the blobbers, sound a little weird that we even used to have something like this before).
cc @guruhubb @dabasov

**Comment by devyetii** [comment]
Here's my results of trying those endpoints using `test.zus.network`, some scenarios are flawed, all will be fixed in 0chain/0chain#2150 and will add system tests for the exact same scenarios.
```
Tested Manually

- /graph-write-price
	- Stake one of the blobbers => Increase
		- zbox sp-lock --blobber_id <blobber_id> --tokens 1
		- Get the blobber and make sure it's staked (`zbox bl-info --blobber_id <blobber_id>`).
		- Make sure `gs.total_staked` is increased with the amount.
		- Make sure `gs.staked_storage` is increased with the value (blobber.total_stake / blobber.write_price)
		- Make sure the new write price is `gs.total_staked / gs.staked_storage`
	- Increase write price of one of the blobbers => Increase
		- zbox bl-update --blobber_id <blobber_id> --write_price XXX
		- Get the blobber and make sure the new write_price is applied
		- Make sure gs.staked_storage is updated with (new.total_stake / new.write_price - old.total_stake / old.write_price)
		- Make sure avg_write_price is gs.total_staked / gs.staked_storage
	- Delete/Kill one of the staked blobbers (todo: when kill/shutdown is merged)
- /graph-total-challenge-pools
	- Upload new file => increase	
	- Cancel allocation => decrease
	- Kill blobber => decrease
	- Blobber penality => decrease (?????)
- /graph-allocated-storage
	- Add new blobber to an allocation => increase
	- Cancel/Finalize allocation => decrease (bug)
	- Reduce allocation's size => decrease
- /graph-used-storage
	- Upload file => increase
	- Remove file => decrease
	- Update the file with a bigger one => increase
	- Update the file with a smaller one => decrease
	- Cancel an allocation => decrease (bug)
- /graph-total-staked
	- Stake a blobber/validator (sp-lock) => increase
	- Stake a miner/sharder (zwallet mn-lock) => increase
	- Unstake a blobber (sp-unlock) => decrease (bug - Move the calculation to provider_aggregate files)
	- Unstake a miner/sharder (sp-unlock) => decrease (bug - Move the calculation to provider_aggregate files)
	- Kill a staked provider => decrease (todo)
- /graph-total-minted
	- Calling `zbox collect_rewards` with a blobber that deserves a reward => increase (bug: missing tokens flag)
	- Unlocking a stake pool the has rewards (sp-unlock) => increase
		- Stake 2 blobbers
		- Add allocation `data 1 parity 1` and make sure it's attached to these 2 blobbers
		- Upload a file there. Make sure rewards are generated for one of the blobbers (sp-info)
		- Stake another blobber
		- `zbox updateallocation --add_blobber --remove_blobber`
		- `zbox sp-unlock <removed_blobber_id>`
	- Finishing/cancelling an allocation => increase
		- Stake 2 blobbers
		- Add allocation `data 1 parity 1` and make sure it's attached to these 2 blobbers
		- Upload a file there. Make sure rewards are generated for one of the blobbers (sp-info)
		- Cancel/Finalize the allocation
		- Unlock stake pools
	- Unlocking a stake pool the has rewards (mn-unlock) => increase
		- Stake 1 miner
		- Unstake again
	- Free-storage allocation => increase
		- Create an allocation
		- Get a free assigner token
		- Updateallocation
	- Bridge Mint (Convert WZCN to ZCN) => increase
		- Test as mentioned by Kishan
- /graph-total-locked
	- Either sum of `total_staked` and `client_locked` or `total_staked * zcn_price` (?????) (todo: Remove `TotalValueLocked` from global snapshot)
- /graph-challenges
	- Upload a file (increase)
	- Kill a blobber with challenges (decrease) (todo)
	- Add a blobber to the allocation with files (increase)
- /graph-token-supply (simply minted + burnet)
	- Same as /graph-total-minted for increase
	- Decrease on Burn
		- Transfer ZCN to WZCN
		- Bruning Fees if config.fees_enabled
- /total-minted
	- Exactly same as /graph-total-minted?data-points=1
- /total-blobber-capacity
	- New blobber added => increase
	- Blobbers killed => decrease (todo)
	- Blobber increased capcaity => increase
	- Blobber decreased capacity => decrease (is this even possible ?)
- /average-write-price
	- Exactly same as /graph-write-price?data-points=1
- /total-staked
	- Exactly Same as /graph-total-staked?data-points=1
- /total-successful-challenges and /total-total-challenges
	- Same as /graph-challenges?data-points=1
- /total-allocated-storage
	- Same as /graph-allocated-storage?data-points=1

=======================================================================================
Per-Blobber
=======================================================================================
- /graph-blobber-service-charge
	- This field, TotalServiceCharge is no more used
- /graph-blobber-challenges-passed
	- Upload a file in an allocation attached to this blobber => increase
- /graph-blobber-challenges-completed
	- Upload a file in an allocation attached to this blobber => increase
- /graph-blobber-inactive-rounds
	- Leave for sometime without attaching to an allocation => increase
	- Attach to an allocation => stop increasing
- /graph-blobber-write-price
	- Increase write price (bl-update) => increase
	- Decrease write price => decrease
- /graph-blobber-capacity
	- Increase capacity (bl-update) => increase
	- Decrease capacity (bl-update) => decrease (is this allowed ??)
- /graph-blobber-allocated
	- Increase allocated (bl-update) => increase
- /graph-blobber-saved-data
	- Upload new file => increase
	- Delete file => decrease
- /graph-blobber-read-data
	- Download a file => increase (bug: will fix in 2148)
- /graph-offers-total
	- Add a new allocation with requirements matches that blobber => increase
	- Cancel allocation => decrease
- /graph-blobber-unstake-total
	- Unlock a stake pool of a provider => increase (bug: will fix in 2148)
- /graph-blobber-unstake-total
	- Stake a blobber/miner/sharder => increase
	- Unstake a blobber/miner/sharder => decrease
- /graph-blobber-challenges-open
	- Add new file => increase
	- Cancel allocation => stop increasing
```
